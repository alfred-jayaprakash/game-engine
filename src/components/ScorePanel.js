import React from 'react';
import {Row, Col} from 'reactstrap';

const ScorePanel = props => {
  return (
    <Row className="text-dark bg-light">
      <Col><h3>Score: {props.score}</h3></Col>
    </Row>
  );
};

export default ScorePanel;
