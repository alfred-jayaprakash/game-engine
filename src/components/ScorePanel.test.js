import React from 'react';
import {render, fireEvent, screen, act, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {shallow} from 'enzyme';
import ScorePanel from './ScorePanel';

let container;
let score;

describe ('Snapshot tests', () => {
  it ('should render correctly without issues', () => {
    const component = shallow (<ScorePanel score={100} />);

    expect (component).toMatchSnapshot ();
  });
});

describe ('Functional tests', () => {
  beforeAll (() => {
    //ARRANGE
  });

  beforeEach (() => {
    container = document.createElement ('div');
    document.body.appendChild (container);
  });

  afterEach (() => {
    document.body.removeChild (container);
    container = null;
  });

  it ('initial screen should load correctly', () => {
    score = 200;
    render (<ScorePanel score={score} />, container);
  });
});
