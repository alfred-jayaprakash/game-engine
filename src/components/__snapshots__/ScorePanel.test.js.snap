// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should render correctly without issues 1`] = `
<Row
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <h5
      className="text-dark bg-light"
    >
      Score: 
      100
    </h5>
  </Col>
</Row>
`;
